src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -305,11 +305,11 @@ static void uct_cuda_copy_event_desc_init(ucs_mpool_t *mp, void *obj, void *chun
 static void uct_cuda_copy_event_desc_cleanup(ucs_mpool_t *mp, void *obj)
 {
     uct_cuda_copy_event_desc_t *base = (uct_cuda_copy_event_desc_t *) obj;
-    int active;
+    CUcontext cuda_context;
 
-    UCT_CUDADRV_CTX_ACTIVE(active);
-
-    if (active) {
+    UCT_CUDA_FUNC_LOG_ERR(cuCtxGetCurrent(&cuda_context));
+    if ((cuda_context != NULL) &&
+        (!memcmp(&cuda_context, &base->cuda_context, sizeof(cuda_context)))) {
```

Comment:
1. shouldn't we save the context during uct_cuda_copy_event_desc_init? or during iface_open?
2. the check if cuda context is the same repeats in multiple places, can add helper inline function?
3. memcmp() looks weird, can use just == ?

Reply:
@yosefe 

> shouldn't we save the context during uct_cuda_copy_event_desc_init? or during iface_open?

We initialize base->cuda_context just after event record in zcopy path [here](https://github.com/openucx/ucx/pull/7679/files#diff-3d338b2219f6061da816a72f3a6cde940f55450c156a252afb3d87a2e550c07fR152).

> the check if cuda context is the same repeats in multiple places, can add helper inline function?

Will do

> memcmp() looks weird, can use just == ?

I tried this at first but thought memcmp is safer in case CUcontext datatype changes?

Reply:
> We initialize base->cuda_context just after event record in zcopy path here.
so maybe compare cuda_context to iface->cuda_context - why need also base->cuda_cotext?

> I tried this at first but thought memcmp is safer in case CUcontext datatype changes?
1. it will not really change IMO; will break Cuda ABI
2. even if does, compiler can compare a primitive type correctly

Reply:
> so maybe compare cuda_context to iface->cuda_context - why need also base->cuda_cotext?

Actually this was how I tried the first time before I posted the PR where I had a field in base_event that had a reference to iface and hence I could compare with iface->cuda_context. It turned out that by the time I mpool_cleanup is called, references to iface->cuda_context was invalid because iface was already 0x0. I'm not sure why this happens. This was I had each event carry a reference to the cuda_context that was used to create it.

Regarding not using memcmp, I'll change to just use ==.

Reply:
so we should call ucs_mpool_cleanup() before removing iface->cuda_context
also, can get iface pointer from "mp" by using ucs_container_of(mp, uct_cuda_copy_iface_t, cuda_event_desc) - no need to save pointer in desc